### PR TITLE
Added Context for Filter Placeholder Text Translation

### LINKED ISSUE #1032: The usage of the "Filter language" message is unclear
This is a question about the message `"language-searcher-placeholder": "Filter language"`

I suspect that it's no actually used anywhere.

I tried finding it in the actual app, but no matter where I clicked, I couldn't see it.

It may be used in `void KiwixChoiceBox::showPlaceholder()`, but that method doesn't seem to be called anywhere. However, the last time I wrote any C++ code was in 2011, so I might be wrong.

If it's not used, it should be removed.

If it used somewhere, it would be great to clarify in qqq what does it mean exactly or perhaps change the message itself. Maybe changing it to "Filter by language" could be clearer.

### PR DESCRIPTION
**Pull request by ShaopengLin**
Fix #1032

### CODE REVIEW

File: resources/i18n/qqq.json [hunk 1]
```diff
@@ -50,5 +50,8 @@
 	"monitor-clear-dir-dialog-title": "\"Monitor\" means \"watch\" in this context. The monitor directory is monitored/watched for new ZIM files.",
 	"monitor-clear-dir-dialog-msg": "\"Monitor\" means \"watch\" in this context. The monitor directory is monitored/watched for new ZIM files.",
 	"open-book": "\"Open\" is a imperative, not an adjective.",
-	"open-previous-tabs-at-startup": "The tabs that were open when the user closed the application is opened again when the application is restarted."
+	"open-previous-tabs-at-startup": "The tabs that were open when the user closed the application is opened again when the application is restarted.",
+	"language-searcher-placeholder": "Placeholder text displayed to indicate the ZIM file searcher will filter by language",
+    "category-searcher-placeholder": "Placeholder text displayed to indicate the ZIM file searcher will filter by category",
+    "content-type-searcher-placeholder": "Placeholder text displayed to indicate the ZIM file searcher will filter by content type"
```
Comment (veloman-yunkan) [inline resources/i18n/qqq.json:56]: Inconsistent indentation (in this file tabs must not be replaced with spaces)

**Review by veloman-yunkan — CHANGES_REQUESTED** [review]
Actually qqq.json has fallen quite behind en.json. Maybe we should fully sync it in one (this) PR? @kelson42

**Comment by kelson42** [comment]
@veloman-yunkan Yes, but it is important that at review time, this point is checked accurately.

**Comment by kelson42** [comment]
@veloman-yunkan Review welcome ;)
@ShaopengLin rebase welcome ;)

**Comment by ShaopengLin** [comment]
Will do rebase after review.

File: resources/i18n/qqq.json [hunk 2]
```diff
@@ -50,5 +50,129 @@
...
+	"content-type":"Describes the type of contents available on a Zim file, for example, text or pictures.",
+	"reset-sort":"Represents the action of resetting the sorting of the items to default order.",
+	"open":"{{Identical|Open}}",
+	"download":"{{Identical|Download}}",
+	"resume":"\"Resume\" here refers to continuing a task.",
+	"port-for-local-kiwix-server-setting":"Describes the port of the computer of which the local Kiwix server will be hosted on.",
+	"zoom-level-setting":"Describes the default zoom level of articles displayed on this desktop application.",
+	"download-directory-setting":"Describes the directory in the file system where Zim files will be downloaded to.",
+	"monitor-directory-setting":"Describes the directory in the file system where we will monitor for Zim file changes.",
+	"monitor-directory-default":"Describes the default directory in the file system where we will monitor for Zim file changes.",
```
Comment (veloman-yunkan) [inline resources/i18n/qqq.json:109]: I think that this is no longer used. We should rather drop such entries from `en.json`

File: resources/i18n/qqq.json [hunk 3]
```diff
@@ -50,5 +50,129 @@
...
+	"reset":"{{Identical|reset}}",
+	"clear":"\"Clear\" here refers to emptying something.",
+	"browse":"{{Identical|Browse}}",
+	"about-kiwix-desktop-title":"Describes the title of this desktop application",
+	"about-kiwix-desktop-description":"Brief description of this desktop applciation.",
```
Comment (veloman-yunkan) [inline resources/i18n/qqq.json:114]: applciation → application

File: resources/i18n/qqq.json [hunk 4]
```diff
@@ -50,5 +50,129 @@
...
+	"about-learn-more-about-kiwix":"Hyperlink text directing to more details about Kiwix.",
+	"about-release-title":"Title text for the section on the release stages of this desktop application.",
+	"about-source-code":"Describes the licensing of the source code used for this desktop application.",
+	"about-version":"Describes the release version of this desktop application.",
+	"about-reporting-problem-title":"Title text for the section for how to report a problem for this desktop applciation.",
```
Comment (veloman-yunkan) [inline resources/i18n/qqq.json:119]: The english text for this entry is buggy. Should be "Reporting problem**s**"

File: resources/i18n/qqq.json [hunk 5]
```diff
@@ -50,5 +50,129 @@
...
+	"about-report-issue":"Description of instructions on how to report an issue.",
+	"about-report-issue-2":"Description of instructions on how to report an issue.",
+	"about-libraries-title":"Title text for the section on what libraries are used in this desktop applciation.",
```
Comment (veloman-yunkan) [inline resources/i18n/qqq.json:122]: applciation → application

File: resources/i18n/qqq.json [hunk 6]
```diff
@@ -50,5 +50,129 @@
...
+	"kiwix-server-running-message":"Text displayed when kiwix server has started running.",
+	"kiwix-server-description":"Description text about capabilities of Kiwix server",
+	"fullscreen-notification":"Notification Text displayed when the desktop applciation is set to fullscreen.",
```
Comment (veloman-yunkan) [inline resources/i18n/qqq.json:125]: applciation -> application

File: resources/i18n/qqq.json [hunk 7]
```diff
@@ -50,5 +50,129 @@
...
+	"online-files":"The Zim files that can be found online.",
```
Comment (veloman-yunkan) [inline resources/i18n/qqq.json:126]: files → flies

File: resources/i18n/qqq.json [hunk 8]
```diff
@@ -50,5 +50,129 @@
...
+	"gutenberg":"The Gutenberg Project",
+	"mooc":"Massive Open Online Courses",
+	"phet":"Physics Education Technology",
+	"psiram":"{{Identical|Psiram}}",
+	"stack_exchange":"{{Identical|Stack Exchange}}",
+	"ted":"{{Identical|Ted}}",
+	"vikidia":"{{Identical|Vikidia}}",
+	"wikihow":"{{Identical|wikiHow}}",
+	"fulltext-search":"Represents the action of searching the text appearances in all the articles in a Zim file.",
+	"ftindex":"Describes the index used to apply to the text to be searched.",
```
Comment (veloman-yunkan) [inline resources/i18n/qqq.json:145]: Change to "Text indicating that fulltext search is supported"

File: resources/i18n/qqq.json [hunk 9]
```diff
@@ -50,5 +50,129 @@
...
+	"details":"A content type for Zim files representing it has more than just an introduction.",
+	"ok":"{{Identical|Ok}}",
+	"no-filter":"No filter selected for searches.",
+	"open-link-new-tab":"Represents the action of opening a hyperlink in an article on in a new tab.",
```
Comment (veloman-yunkan) [inline resources/i18n/qqq.json:149]: "on in" → in

File: resources/i18n/qqq.json [hunk 10]
```diff
@@ -50,5 +50,129 @@
...
+	"download-dir-dialog-title":"A question to confirmation the action to change the download directory.",
```
Comment (veloman-yunkan) [inline resources/i18n/qqq.json:150]: confirmation → confirm

File: resources/i18n/qqq.json [hunk 11]
```diff
@@ -50,5 +50,129 @@
...
+	"invalid-port":"Error text displayed when the port selected is invalid in the local system.",
+	"zim-open-fail-title":"Error title text for the situation when opening a file as a Zim file failed.",
+	"zim-open-fail-text":"Error description text for when opening a file as a Zim file failed.",
+	"monitor-directory-tooltip":"Description text on what the monitor directory does.",
+	"next-tab":"Represents the action of switching to the next tab with respect to the current tab.",
+	"previous-tab":"Represents the action of switching to the previous tab with respect to the current tab.",
+	"cancel-download": "Represents the action of cancelling an on-going download of a Zim file.",
+	"cancel-download-text": "A question to confirmation the action to cancel the download of a Zim file.",
```
Comment (veloman-yunkan) [inline resources/i18n/qqq.json:158]: confirmation → confirm

File: resources/i18n/qqq.json [hunk 12]
```diff
@@ -50,5 +50,129 @@
...
+	"delete-book": "Represents the action of deleting an existing Zim file.",
+	"delete-book-text": "A question to confirmation the action to delete an existing Zim file.",
```
Comment (veloman-yunkan) [inline resources/i18n/qqq.json:160]: confirmation → confirm

File: resources/i18n/qqq.json [hunk 13]
```diff
@@ -50,5 +50,129 @@
...
+	"download-storage-error": "Error title text displayed when something is wrong the directory of storage for Zim files",
+	"download-storage-error-text": "Error description text for when something is wrong the directory of storage for Zim files.",
```
Comment (veloman-yunkan) [inline resources/i18n/qqq.json:162]: ... wrong **with** the directory ...

File: resources/i18n/qqq.json [hunk 14]
```diff
@@ -50,5 +50,129 @@
...
+	"download-unavailable": "Error title text displayed when downloading Zim files are not available.",
+	"download-unavailable-text": "Error description text displayed when downloading Zim files are not available.",
```
Comment (veloman-yunkan) [inline resources/i18n/qqq.json:164]: are → is

**Review by veloman-yunkan — CHANGES_REQUESTED** [review]
I think that I physically felt how you got bored as you documented all those strings. I can only console you that now you are not alone with respect to such perception of this PR :)

I'd be glad to have it merged right away, yet there are a few things that should be corrected. @kelson42 @Popolechien Can we have [The Boring Company](https://www.boringcompany.com/) fund this PR?

File: resources/i18n/qqq.json (hunk 7 above)
Comment (ShaopengLin) [reply resources/i18n/qqq.json:126]: This one is actually not a typo. I can feel your eye strain rn.

Comment (veloman-yunkan) [reply resources/i18n/qqq.json]: :+1: This was actually a test that you don't take my feedback blindly

**Review by veloman-yunkan — COMMENTED** [review]
If the volume of changes made after a review is small compared to the volume of changes already reviewed please introduce them (the new modifications) as fixup commits (so that reviewers' task is made easier).

I have made a few smaller corrections to your PR in a fixup commit (since in this particular case it amounts to less total effort compared to writing comments and having you address them).

Please rebase and squash the fixup commit to get the formal approval to merge this PR.

**Comment by amire80** [comment]
Thank you! It may feel boring, but it's enormously helpful to the volunteer translators!

**Pull request merged**